lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -2695,6 +2695,118 @@ class ConvertAtenPermuteOp : public OpConversionPattern<AtenPermuteOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenSliceTensorOp : public OpConversionPattern<AtenSliceTensorOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenSliceTensorOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    Location loc = op.getLoc();
+    TypeConverter *typeConverter = getTypeConverter();
+
+    auto input = adaptor.self();
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    RankedTensorType resultType =
+        typeConverter->convertType(op->getResult(0).getType())
+            .cast<RankedTensorType>();
+    int64_t resultRank = resultType.getRank();
+    Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    Value one = rewriter.create<arith::ConstantIndexOp>(loc, 1);
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return op->emitError("unimplemented: dim is not constant");
+
+    SmallVector<Value> inputShape = getTensorSizes(rewriter, loc, input);
+    Value dimSize = inputShape[dim];
+
+    auto adjustStartOrEnd = [](Location loc, Value dimSize, Value &startOrEnd,
```

Comment:
using [&] could capture the variable from context and `Location loc, Value dimSize, ConversionPatternRewriter &rewriter` can be used directly rather than passing as arguments. The logic to check NoneType can also be wrapped into the lambda. The lambda become
```
auto adjustStartOrEnd = [&](Value startOrEndTorch, Value startOrEndBuiltIn, Value valueForNone) {
  if (startOrEndTorchType.getType().isa<NoneType>())
      return valueForNone;
...
}
```